src/spikeinterface/widgets/traces.py

```diff
@@ -114,6 +120,15 @@ def __init__(
         else:
             channel_locations = None
 
+        if not rec0.has_time_vector():
+            self.times = None
+            t_start = 0
+            t_end = rec0.get_duration(segment_index=segment_index)
+        else:
+            self.times = rec0.get_times(segment_index=segment_index)
+            t_start = self.times[0]
+            t_end = self.times[-1]
```

Comment:
This will not work when you have multiple segment and you use ipywidget because you can switch segment with a dropdone menu
We should have a list instead.
Am I wrong ?
Did you try with multi segment and several t_start and or times vector ?

Reply:
And more importantly we should not have attribute setting sin the init. Everything should be done with the data_plot.